### PR TITLE
Fixes tweaselORG/meta#30: Creation of Android emulators

### PR DESCRIPTION
**Pull request by baltpeter**

### CODE REVIEW

File: src/emulator.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+ * For choosing the system image you can either:
+ *
+ * - Specify only `package`, or…
+ * - Specify `apiLevel`, `variant`, and `architecture` but not `package`. Note that not all combinations of these
+ *   properties actually have a system image available.
+ */
+export type EmulatorOptions = MergeExclusive<
+    {
+        /**
+         * The package name of the system image to use for the emulator (as understood by `sdkmanager`.
```
Comment (zner0L) [inline src/emulator.ts]:
```suggestion
         * The package name of the system image to use for the emulator (as understood by `sdkmanager`).
```

**Review by zner0L — CHANGES_REQUESTED** [review]

**Comment by zner0L** [comment]
Creating the emulator runs without any errors, but if I try to run the emulator using:

```js
await runAndroidDevTool('emulator', ['-avd', 'test']);
``` 

I get an error:

```zsh
$ JAVA_HOME=$(/usr/libexec/java_home) tsx test.tmp.ts
(node:8222) UnhandledPromiseRejectionWarning: Error: Command failed with exit code 1: /Users/zner0L/Library/Caches/andromatic/emulator/emulator -avd test
WARNING | platforms subdirectory is missing under /Users/zner0L/Library/Caches/andromatic, please install it
WARNING | ANDROID_SDK_ROOT is missing.
WARNING | platforms subdirectory is missing under /Users/zner0L/Library/Caches/andromatic, please install it
WARNING | invalid sdk root /Users/zner0L/Library/Caches/andromatic
PANIC: Cannot find AVD system path. Please define ANDROID_SDK_ROOT
INFO    | Android emulator version 32.1.13.0 (build_id 10086546) (CL:N/A)
INFO    | checking ANDROID_HOME for valid sdk root.
INFO    | checking ANDROID_SDK_ROOT for valid sdk root.
emulator: WARN: Cannot find valid sdk root from environment variable ANDROID_HOME nor ANDROID_SDK_ROOT,Try to infer from emulator's path
INFO    | guessed sdk root is /Users/zner0L/Library/Caches/andromatic
emulator: WARN: Cannot find valid sdk root path.
```

**Comment by baltpeter** [comment]
Oh, that's fun. Apparently, it needs a `platform` directory but doesn't actually do anything with it. If I delete my existing `platforms` folder, I get the error. If I create an empty `platforms` folder, it works.

**Review by zner0L — CHANGES_REQUESTED** [review]
I deleted the cache folder again to test if it would work on a new machine. Turns out it needs an empty `platform-tools` folder as well:

```
(node:8834) UnhandledPromiseRejectionWarning: Error: Command failed with exit code 1: /Users/zner0L/Library/Caches/andromatic/emulator/emulator -avd test4
WARNING | platform-tools subdirectory is missing under /Users/zner0L/Library/Caches/andromatic, please install it
WARNING | ANDROID_SDK_ROOT is missing.
WARNING | platform-tools subdirectory is missing under /Users/zner0L/Library/Caches/andromatic, please install it
WARNING | invalid sdk root /Users/zner0L/Library/Caches/andromatic
PANIC: Cannot find AVD system path. Please define ANDROID_SDK_ROOT
INFO    | Android emulator version 32.1.13.0 (build_id 10086546) (CL:N/A)
INFO    | checking ANDROID_HOME for valid sdk root.
INFO    | checking ANDROID_SDK_ROOT for valid sdk root.
emulator: WARN: Cannot find valid sdk root from environment variable ANDROID_HOME nor ANDROID_SDK_ROOT,Try to infer from emulator's path
INFO    | guessed sdk root is /Users/zner0L/Library/Caches/andromatic
emulator: WARN: Cannot find valid sdk root path.
```

If that one exists as well, the emulator starts as expected.

**Review by zner0L — APPROVED** [review]
Great, it works now!

**Pull request merged**